src/Microsoft.AspNetCore.Razor.Evolution/Legacy/CSharpCodeParser.cs

```diff
@@ -643,12 +648,15 @@ private void SectionBlock(string left, string right, bool caseSensitive)
         private void NestedBlock()
         {
             Output(SpanKind.Code);
+
             var wasNested = IsNested;
             IsNested = true;
             using (PushSpanConfig())
             {
                 ParseBlock();
             }
+
+            Span.Start = CurrentLocation;
             Initialize(Span);
```

Comment:
Every time you call Initialize it looks like you're setting `Span.Start` prior. Any reason to not throw that into the Initialize method? Maybe i'm missing something.